Implement Control Protocol v2

fsat opened the pull request:
Fixes https://github.com/typesafehub/conductr/issues/925.

- [x] Rename v1.1 Control Protocol to v2
- [x] Present `bundle.conf` found within bundle zip file as part of the `conduct load` request as opposed to parsing the `bundle.conf` for fields declared within the file.
- [x] Present the overlay `bundle.conf` if found within the configuration zip in the similar fashion

huntc commented:
Looking good

huntc commented on src/main/scala/com/typesafe/conductr/client/ConductRController.scala:
```diff
@@ -192,6 +210,13 @@ object ConductRController {
+    val result = IO.unzip(new File(absolute(zipFile).path.toString()), destDir, { fileName: String => fileName.contains(entryName) }).head
+    Uri(result.getAbsolutePath)
+  }
+
```
You should do a quick check to see whether the native packager already includes such a function - I'm unsure, but it could be worth looking.

markusjura replied:
Also we have a similar logic already in the `ConductR.scala` file in which we unzip the bundle as well and access the `bundle.conf` file: https://github.com/fsat/sbt-conductr/blob/ctrl-protocol-v2/src/main/scala/com/typesafe/conductr/sbt/ConductR.scala#L41-L45

Maybe this logic can be combined?

huntc commented on src/test/scala/com/typesafe/conductr/sbt/ConductRSpec.scala:
```diff
@@ -0,0 +1,362 @@
+      }
+    })
+
+}
```
I like these additional tests

markusjura commented:
LGTM (Beside my comment with unzipping the bundle file).

markusjura commented:
Nice to have additional tests here.

fsat commented:
@markusjura - I've implemented your feedback on consolidating the unzipping of the bundle file. I have also added additional tests around the load functionality. Can you review again please?

I have squashed the commits as well, and everything is ready to merge.

cc @huntc

markusjura commented:
LGTM

fsat commented:
Latest change was to rename body part `bundleOverlayConf` to `bundleConfOverlay` - manual tested successfully against ConductR master branch.

fsat commented:
Also tested successfully against ConductR 1.0 branch.
